code/modules/vending/_vending.dm

```diff
@@ -1149,18 +1155,21 @@ GLOBAL_LIST_EMPTY(vending_products)
 	if(id_card?.registered_account && id_card.registered_account == linked_account)
 		return TRUE
 
-/obj/machinery/vending/custom/canLoadItem(obj/item/I, mob/user)
+/obj/machinery/vending/custom/canLoadItem(obj/item/item_to_insert, mob/user)
 	. = FALSE
-	if(I.flags_1 & HOLOGRAM_1)
+	if(item_to_insert in blacklisted_items)
```

Comment:
Not even mentioning that I don't think this works, because you're checking for a reference in a list of types